src/pharmpy/tools/estmethod/algorithms.py

```diff
@@ -146,22 +187,50 @@ def _create_base_model(model):
     while len(base_model.estimation_steps) > 0:
         base_model = remove_estimation_step(base_model, 0)
 
-    base_model = add_estimation_step(base_model, **est_settings)
-    base_model = add_estimation_step(base_model, **eval_settings)
+    if add_eval_after_est:
+        base_model = add_estimation_step(base_model, **est_settings)
+        base_model = add_estimation_step(base_model, **eval_settings)
+    else:
+        base_model = add_estimation_step(base_model, **est_settings)
+
     return base_model
 
 
 def _create_candidate_model(
-    method, solver, parameter_uncertainty_method, update_inits, only_evaluation, model
+    method,
+    solver,
+    parameter_uncertainty_method,
+    add_eval_after_est,
+    update_inits,
+    only_evaluation,
+    model,
 ):
-    est_settings = _create_est_settings(method, only_evaluation)
     laplace = True if method == 'LAPLACE' else False
-    eval_settings = _create_eval_settings(laplace, parameter_uncertainty_method)
 
-    eval_method = eval_settings['method']
+    if not only_evaluation:
+        est_settings = _create_est_settings(
+            method, parameter_uncertainty_method, add_eval_after_est
+        )
+    else:
+        est_settings = None
+
+    if only_evaluation or add_eval_after_est:
+        eval_settings = _create_eval_settings(laplace, parameter_uncertainty_method)
+        eval_method = eval_settings['method']
+    else:
+        eval_settings = None
+        eval_method = None
+
+    if only_evaluation:
+        all_estrec_methods = [eval_method]
+    elif add_eval_after_est:
+        all_estrec_methods = [method, eval_method]
+    else:
+        all_estrec_methods = [method]
+
```

Comment:
This logic is a bit confusing to me, especially in combination with the code on 243-250. So we have three possible cases: 

1. Only evaluation
2. Estimation and evaluation (relevant when assessing `methods`)
3. Only estimation (relevant when not assessing `methods`)

To me it would be a little easier to read if all of these things were set in one `if-else` block, since e.g. the description could be changed afterwards.

Reply:
Changed, see if the logic is easier to understand now.